### PR TITLE
details on weekday argument for DateOffset docs

### PR DESCRIPTION
**Pull request by Detoro**
- [ ] towards #52650
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.

### CODE REVIEW

**Comment by raj-thapa** [comment]
Hey @Detoro . Can you try commenting ``` pre-commit.ci autofix ```?

File: pandas/_libs/tslibs/offsets.pyx
Comment (Detoro) [inline pandas/_libs/tslibs/offsets.pyx:1]: pre-commit.ci autofix

**Comment by Detoro** [comment]
pre-commit.ci autofix

**Comment by Detoro** [comment]
pre-commit.ci autofix

**Comment by Detoro** [comment]
pre-commit.ci autofix

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -1373,6 +1373,24 @@ class DateOffset(RelativeDeltaOffset, metaclass=OffsetMeta):
...
         Temporal parameter that add to or replace the offset value.
+    weekday : int {0, 1, ..., 6}, default 0
+        A specific integer for the day of the week.
+        - 0 is Monday
+        - 1 is Tuesday
+        - 2 is Wednesday
+        - 3 is Thursday
+        - 4 is Friday
+        - 5 is Saturday
+        - 6 is Sunday.
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/offsets.pyx:1384]: this doesn't look quite right, you need a blank line before "A specific integer"

image

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for your PR

this addresses part of the issue - the other part is about the link

in `See also`, could you replace `dateutil.relativedelta.relativedelta` with https://dateutil.readthedocs.io/en/stable/relativedelta.html please?

thanks

File: pandas/_libs/tslibs/offsets.pyx (hunk 1 above)
Comment (keelancosgrove) [reply pandas/_libs/tslibs/offsets.pyx:1384]: Working with @Detoro on this - we'd like to view the built HTML but have been encountering some issues.
Namely, after running `python make.py --single pandas.tseries.offsets.DateOffset` in the `doc` directory, we don't see any output in directory `reference/API` for DateOffset. We're getting this warning from Sphinx:

`/mnt/c/Users/batiu/OneDrive/Desktop/pandasContribution/pandas-toro/doc/source/index.rst:31: WARNING: autosummary references excluded document 'pandas.tseries.offsets.DateOffset'. Ignored.`

Are we building this class correctly, and if so why is the class's documentation being excluded by "autosummary references"?

**Comment by keelancosgrove** [comment]
> thanks for your PR
> 
> this addresses part of the issue - the other part is about the link
> 
> in `See also`, could you replace `dateutil.relativedelta.relativedelta` with https://dateutil.readthedocs.io/en/stable/relativedelta.html please?
> 
> thanks

We were planning to do this in a follow up PR. Do you prefer if both parts are done in separate PRs, or a single PR?

**Comment by Detoro** [comment]
pre-commit.ci autofix

File: pandas/_libs/tslibs/offsets.pyx (hunk 1 above)
Comment (keelancosgrove) [reply pandas/_libs/tslibs/offsets.pyx]: @MarcoGorelli PR is updated, can you take another look? We'll squash the commits as well.

**Comment by keelancosgrove** [comment]
@mroeschke could you take a look?

File: pandas/_libs/tslibs/offsets.pyx [hunk 2]
```diff
@@ -1467,6 +1467,27 @@ class DateOffset(RelativeDeltaOffset, metaclass=OffsetMeta):
         previous midnight.
     **kwds
         Temporal parameter that add to or replace the offset value.
+    weekday : int {0, 1, ..., 6}, default 0
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/offsets.pyx:1470]: can this go before `kwds`? that should typically be the last argument

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks, this is close

if this doesn't fully close the issue, can you replace `closes` with `towards` in the pull request description please?

File: pandas/_libs/tslibs/offsets.pyx (hunk 2 above)
Comment (Detoro) [reply pandas/_libs/tslibs/offsets.pyx:1468]: Done.

**Review by MarcoGorelli — APPROVED** [review]
thanks @Detoro , lgtm on green

**Comment by MarcoGorelli** [comment]
why did you close this?

**Comment by Detoro** [comment]
my bad @MarcoGorelli

**Pull request merged**